signalr-go-server/pkg/signalr/websocketServer.go

```diff
@@ -0,0 +1,150 @@
+package signalr
+
+import (
+	"bytes"
+	"crypto/rand"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"golang.org/x/net/websocket"
+	"net/http"
+	"sync"
+	"time"
+)
+
+// MapHub used to register a SignalR Hub with the specified ServeMux
+func MapHub(mux *http.ServeMux, path string, hubPrototype HubInterface) {
+	mux.HandleFunc(fmt.Sprintf("%s/negotiate", path), negotiateHandler)
+	server := NewServer(hubPrototype)
+	mux.Handle(path, websocket.Handler(func(ws *websocket.Conn) {
+		connectionID := ws.Request().URL.Query().Get("id")
+		if len(connectionID) == 0 {
+			// Support websocket connection without negotiate
+			connectionID = getConnectionID()
+		}
+		if protocol, err := processHandshake(ws); err != nil {
```

Comment:
This is at the wrong layer. The handshake needs to be processed on the connection abstraction not directly at the websocket layer.

Reply:
See [4744552](https://github.com/davidfowl/signalr-ports/pull/6/commits/47445523e21b7fdf5fc000f75c18bcfd51b362dc)